ArduCopter/control_flip.cpp

```diff
@@ -199,9 +201,9 @@ void Copter::flip_run()
         // check for successful recovery
         if (fabsf(recovery_angle) <= FLIP_RECOVERY_ANGLE) {
             // restore original flight mode
-            if (!set_mode(flip_orig_control_mode)) {
+            if (!set_mode(flip_orig_control_mode, flip_orig_control_mode_reason)) {
```

Comment:
I understand the reason for this but I don't agree with it. Imagine the previous mode was AltHold and was changed from GCS, then you enter Flip mode from your Tx, you would then be back to AltHold and log would show it was changed from GCS when that's not true.
Even worse is that if you can't enter the previous mode you enter Stabilize but in that case you say that it was reason unknown. For me, these are the exact same case, the reason for change is the simple flight control.